crates/ruma-common/src/push/condition.rs

```diff
@@ -1023,6 +1124,90 @@ mod tests {
         let sender_notification_permission =
             PushCondition::SenderNotificationPermission { key: NotificationPowerLevelsKey::Room };
 
-        assert!(sender_notification_permission.applies(&first_event, &context));
+        assert!(sender_notification_permission.applies(&first_event, &context).await);
+    }
+
+    #[cfg(feature = "unstable-msc4306")]
+    #[apply(test!)]
+    async fn thread_subscriptions_match() {
+        use crate::{event_id, EventId};
+
+        let context = push_context().with_has_thread_subscription_fn(|event_id: &EventId| {
+            Box::pin(async move {
+                // Simulate thread subscriptions for testing.
+                event_id == event_id!("$subscribed_thread")
+            })
+        });
+
+        let subscribed_thread_event = FlattenedJson::from_raw(
+            &serde_json::from_str::<Raw<JsonValue>>(
+                r#"{
+                "event_id": "$thread_response",
```

Comment:
All these string literals end one indentation level less than they started. Looks like all lines except the first should be indented one level.